src/transformers/models/clip/modeling_clip.py

```diff
@@ -217,6 +217,13 @@ def forward(
         inputs_embeds: Optional[torch.FloatTensor] = None,
     ) -> torch.Tensor:
         seq_length = input_ids.shape[-1] if input_ids is not None else inputs_embeds.shape[-2]
+        max_position_embedding = self.position_embedding.weight.shape[0]
+
+        if seq_length > max_position_embedding:
```

Comment:
* No need to create a new variable if it's only used once
* We should instead store a variable from the config to avoid expensive checks on the weight size for every forward pass

```suggestion
        if seq_length > self.max_position_embeddings:
```

And in the init

```
    def __init__(self, config: CLIPTextConfig):
        super().__init__()
        embed_dim = config.hidden_size
        self.max_position_embeddings = config.max_position_embeddings
```

Reply:
I realised I asked for this change -- coming back to it, I think that we shouldn't set `self.max_position_embeddings` as the user can change the embeddings with `set_input_embeddings` --> the value from the config will become out of date. In this case, checking the shapes as before is less brittle, could you change it back to this? Apologies for flip-flopping here.